docs/dev/table/streaming/temporal_tables.md

```diff
@@ -22,9 +22,13 @@ specific language governing permissions and limitations
 under the License.
 -->
 
-Temporal Tables represent a concept of a (parameterized) view on a changing history table that returns the content of a table at a specific point in time.
+Temporal tables represent a concept of changing table that its data is deemed to be effective or valid along some time period.
 
-Flink can keep track of the changes applied to an underlying append-only table and allows for accessing the table's content at a certain point in time within a query.
+The changing table can either be a changing history table which tracks the changes (e.g. database changelogs) or a changing dimension table which materializes the changes (e.g. database tables).
+
+For the changing history table, Flink can keep track of the changes and allows for accessing the content of the table at a certain point in time within a query. In Flink, this kind of table is represented by a *Temporal Table Function*.
+
+For the changing dimension table, Flink allows for accessing the content of the table at processing time within a query. In Flink, this kind of table is represented by a *Temporal Table*. In the future, a *Temporal Table* can also represents a changing history table.
```

Comment:
Mention that results are not deterministic/not replayable. Remove the future sentence for now. This is future work.